lapack/gonum/dlassq.go

```diff
@@ -77,21 +77,25 @@ func (impl Implementation) Dlassq(n int, x []float64, incx int, scale float64, s
 		ax := scale * math.Sqrt(sumsq)
 		switch {
 		case ax > dtbig:
-			// We assume scale >= sqrt( TINY*EPS ) / dsbig, that is, if the
-			// scaled sum is big then its scaling factor should not be too
-			// small.
-			v := scale * dsbig
-			abig += (v * v) * sumsq
+			if scale > 1 {
+				scale *= dsbig
+				abig += scale * (scale * sumsq)
```

Comment:
I take it the order of operations is to prevent an overflow?

Reply:
I think it's rather to prevent underflow in `scale*scale`, after scaling down (by `dsbig`) `scale` can be very small.